### DISCUSSION TITLE
Support for *.disc files ?

### DISCUSSION THREAD

**Discussion by Philmo67** [seerr-team/seerr#581] [Q&A]
Hello,
I materialize the physical DVDs and Blurays, which I have not yet ripped, as *.disc files, in the directories indexed by Jellyfin (following the conventions defined for KODI, here https://kodi.wiki/view/Media_stubs ).
This way, my entire collection is indexed without a hitch by Jellyfin and, either Jellyfin or KODI, when you want to watch the film, tells you to insert the disc in question into the disc drive.
However, Jellyseerr does not consider these films to be available.
I can, of course, manually tag these films as available when someone asks for them, but is there a way for Jellyseerr to take these films, which are present in Jellyfin, into account?

**Comment by Philmo67** [comment]
Seems to be addressed by #522 (or not ?)

**Reply by fallenbagel** [reply]
Does the nfo contain streamdetails? If it contains it will mark them no matter what extension. #522 is for automatic purging. Not related

**Comment by Philmo67** [comment]
```
<fileinfo>
    <streamdetails />
  </fileinfo>
```
so no streamdetail.
Maybe there's something to investigate in the *.disc file so that Jellyfin can insert elements about the stream in the nfo file. 
Thanks for the hint.

**Reply by fallenbagel** [reply]
that is why disc files arent being read. Jellyseerr looks at the resolution

**Reply by Philmo67** [reply]
According to my tests, Jellyfin needs a real video file to write a resolution to the nfo file in a durable way. Too bad the mere presence of the movie in the Jellyfin library isn't enough for Jellyseerr.

**Reply by fallenbagel** [reply]
jellyseerr needs to check resolution to determine whether its a 4k available badge or normal badge.

**Reply by Philmo67** [reply]
Would there be a way to automate something ?

**Reply by fallenbagel** [reply]
> Would there be a way to automate something ?

If you store nfo files with the media, jellyfin will usually write down the streamdetails. So you could write a script to play each media to get the streamdetails. Or add bogus stream details (might face complications but in my testings jellyfin replaced any bogus stream details with the correct ones when you play)

**Reply by Philmo67** [reply]
But *.disc files are not playable, they trigger a popup telling you to play the physical media (and KODI is able to read title and message tags in them so you can tell the physical disc label and where to find the media for example). 
I will try to add bogus stream details using external tools in the nfo files, hoping they won't get destroyed later.

Subsidiary question arising : I also have 3D bluray media (ripped and physical), interpreted and displayed as such by Jellyfin. 
You have to use the adequate 3D naming convention here : https://jellyfin.org/docs/general/server/media/movies/
It seems the 3D "option" is not taken into account by Jellyseerr at all, right ?

**Reply by fallenbagel** [reply]
> Subsidiary question arising : I also have 3D bluray media (ripped and physical), interpreted and displayed as such by Jellyfin. 
> You have to use the adequate 3D naming convention here : https://jellyfin.org/docs/general/server/media/movies/
> It seems the 3D "option" is not taken into account by Jellyseerr at all, right ?
> 

We dont track if a media is 3D or not.

**Comment by Philmo67** [comment]
I've just noticed that when I do the same with *.disc files for my TV shows, Jellyseer automatically detects their presence in Jellyfin and suggests "Play on Jellyfin" (triggering a popup telling you to play the physical media). 
Perhaps here would be a hint to do something similar for movies?

**Reply by fallenbagel** [reply]
The issue is the streamDetails missing. In movies we use the stream details to determine whether 4k or not before registering it as available or not.

In case of shows, there's no streamdetails for tv show top level. So we can determine if it's available or not before checking individual episodes for its width

**Comment by Philmo67** [comment]
I've been unable to get the dummy streamdetails information added to the *.disc files related *.nfo files taken into account (full library scan on the Jellyfin side and restart of various sync tasks on the Jellyseerr side). 
I use *arrs as a rip renaming tool, so movies present as *.disc files on the Jellyfin side are absent from radarr, could this also be part of the problem? (This is also the case for TV shows in sonarr, but as mentioned above, Jellyseerr takes them into account).

**Reply by fallenbagel** [reply]
No the media does not need to be on radarr/sonarr. As mentioned earlier, TV shows work because they take the streamdetails into account after marking the show; when checking the episode resolutions.

As for movies there's no parent (in case of shows there's a hierarchy, shows > seasons > episodes [where streamdetails are]), tv shows gets marked available due to that. I guess we could add logic so that if the resolution is not found, just mark it as available for movies 🤔

**Reply by Philmo67** [reply]
That would make sense : the movie is available if a file supported by the media server is available

**Comment by Philmo67** [comment]
Hello @fallenbagel, sorry to bump this old thread, but I'm still running into the same issue and wanted to check if there's any update.
I noticed a similar (newer) issue that might be related: #2917 ("Emby .strm Movie scan import fails if media not yet probed") and its PR #2918. Sounds like it could be the same root cause as what we talked about here — maybe worth linking the two?
If I understood correctly what you explained back in 2024: for movies, Seerr needs streamDetails/MediaSources to know if it's available, and .disc files never have that info (same for .strm files that were never played). You mentioned a possible fix — just marking it as available anyway if no resolution is found — which would make movies work like shows already do.
Not sure if this is something that could be picked up again? From what I could see in processJellyfinMovie, isn't the check something like:
```
if (hasOtherResolution || (!this.enable4kMovie && has4k)) {
  await this.processMovie(...)
}
```
Would adding a check for "no MediaSources at all" (something like `hasNoStreamData = !metadata.MediaSources?.length)` alongside `hasOtherResolution` work as a quick win?

Happy to test anything if needed, just let me know!
